packages/core/strapi/lib/services/entity-service/types/index.d.ts

```diff
@@ -18,24 +40,118 @@ export interface EntityService {
       | 'publicationState'
       | 'plugin'
     >
-  ): Promise<Attribute.GetValues<TContentTypeUID>[]>;
+  ): Promise<Result<TContentTypeUID>[]>;
+
   findOne<TContentTypeUID extends Common.UID.ContentType>(
     uid: TContentTypeUID,
     entityId: number,
-    params?: EntityService.Params.Pick<TContentTypeUID, 'fields' | 'populate'>
-  ): Promise<Attribute.GetValues<TContentTypeUID> | null>;
+    params?: Params.Pick<TContentTypeUID, 'fields' | 'populate'>
+  ): Promise<Result<TContentTypeUID> | null>;
+
   delete<TContentTypeUID extends Common.UID.ContentType>(
     uid: TContentTypeUID,
     entityId: number,
-    params?: EntityService.Params.Pick<TContentTypeUID, 'fields' | 'populate'>
-  ): Promise<Attribute.GetValues<TContentTypeUID> | null>;
+    params?: Params.Pick<TContentTypeUID, 'fields' | 'populate'>
+  ): Promise<Result<TContentTypeUID> | null>;
+
   create<TContentTypeUID extends Common.UID.ContentType>(
     uid: TContentTypeUID,
-    params?: EntityService.Params.Pick<TContentTypeUID, 'data' | 'files' | 'fields' | 'populate'>
-  ): Promise<Attribute.GetValues<TContentTypeUID>>;
+    params?: Params.Pick<TContentTypeUID, 'data' | 'files' | 'fields' | 'populate'>
+  ): Promise<Result<TContentTypeUID>>;
+
   update<TContentTypeUID extends Common.UID.ContentType>(
     uid: TContentTypeUID,
     entityId: number,
-    params?: EntityService.Params.Pick<TContentTypeUID, 'data' | 'files' | 'fields' | 'populate'>
-  ): Promise<Attribute.GetValues<TContentTypeUID>>;
+    params?: Params.Pick<TContentTypeUID, 'data' | 'files' | 'fields' | 'populate'>
+  ): Promise<Result<TContentTypeUID>>;
+
+  findPage<TContentTypeUID extends Common.UID.ContentType>(
+    uid: TContentTypeUID,
+    params?: Params.Pick<
+      TContentTypeUID,
+      'fields' | 'filters' | 'pagination' | 'sort' | 'populate' | 'publicationState' | 'plugin'
+    >
+  ): Promise<PaginatedResult<TContentTypeUID>>;
+
+  clone<TContentTypeUID extends Common.UID.ContentType>(
+    uid: TContentTypeUID,
+    cloneId: number,
+    params?: Params.Pick<TContentTypeUID, 'fields' | 'populate' | 'data' | 'files'>
+  ): Promise<Result<TContentTypeUID> | null>;
+
+  deleteMany<TContentTypeUID extends Common.UID.ContentType>(
+    uid: TContentTypeUID,
+    params: Params.Pick<TContentTypeUID, 'populate'>
```

Comment:
Similar to my comment on `count`, technically most of these methods accept all the parameters even if they don't use them. For example, `sort` and `fields` are both accepted on `deleteMany` (and I think even used in the query, but that's a bug IMO and they should be stripped out for more efficient queries).

We need to decide if we should include those "useless" params in the types or force users to "clean" or recompose their params between calls -- and follow up with a change in the actual implementation of the methods.

As an example:

- a user does a query with certain params for a findMany
- checks some data in it to see if they should be deleted
- reuses their params to call deleteMany to ensure the same items are deleted is not the best behavior
  - "user friendly" in the sense that they write less code, but unsafe because (unless it's in a transaction) new entities could have been added. So we should encourage them to 

IMO for now we should stick with the actual implementation and include all the parameters, because they do in fact accept everything (and it just occurred to me that it's also probably possible to delete a page of data with 'pagination'), but I'm curious what everyone else thinks.

Reply:
Following a discussion on discord, we will implement the types with the parameters that actually make sense for the method, even if in practice they are able to accept more that are unused.